components/user-mgt/org.wso2.carbon.identity.user.action/src/main/java/org/wso2/carbon/identity/user/action/listener/ActionUserOperationEventListener.java

```diff
@@ -0,0 +1,95 @@
+import org.wso2.carbon.identity.user.action.constant.UserActionConstants;
+import org.wso2.carbon.identity.user.action.factory.UserActionExecutorFactory;
+import org.wso2.carbon.identity.user.action.model.UserActionContext;
+import org.wso2.carbon.user.core.UserStoreClientException;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+/**
+ * This class is responsible for handling the action invocation related to user flows.
+ */
+public class ActionUserOperationEventListener extends AbstractIdentityUserOperationEventListener {
+
+    @Override
+    public int getExecutionOrderId() {
+
+        int orderId = getOrderId();
+        if (orderId != IdentityCoreConstants.EVENT_LISTENER_ORDER_ID) {
+            return orderId;
+        }
+        return 10000;
+    }
+
+    @Override
+    public boolean doPreUpdateCredentialByAdminWithID(String userName, Object credential,
+                                                      UserStoreManager userStoreManager) throws UserStoreException {
+
+        if (!isEnable()) {
+            return true;
+        }
+
+        try {
+            UserActionContext userActionContext = new UserActionContext.Builder()
+                    .userId(userName)
+                    .password((StringBuffer) credential)
+                    .userStoreDomain(UserCoreUtil.getDomainName(userStoreManager.getRealmConfiguration()))
+                    .build();
+
+            ActionExecutionStatus<?> executionStatus =
+                    UserActionExecutorFactory.getUserActionExecutor(ActionType.PRE_UPDATE_PASSWORD)
+                            .execute(userActionContext,
+                                    IdentityContext.getThreadLocalCarbonContext().getTenantDomain());
+
+            if (executionStatus.getStatus() == ActionExecutionStatus.Status.SUCCESS) {
+                return true;
+            } else if (executionStatus.getStatus() == ActionExecutionStatus.Status.FAILED) {
+                Failure failedResponse = (Failure) executionStatus.getResponse();
+                String error = buildErrorMessage(failedResponse.getFailureReason(),
+                        failedResponse.getFailureDescription());
+                throw new UserStoreClientException(error, UserActionConstants.PRE_UPDATE_PASSWORD_ACTION_ERROR_CODE);
+            } else if (executionStatus.getStatus() == ActionExecutionStatus.Status.ERROR) {
+                Error errorResponse = (Error) executionStatus.getResponse();
+                String error = buildErrorMessage(errorResponse.getErrorMessage(), errorResponse.getErrorDescription());
+                throw new UserStoreException(error, UserActionConstants.PRE_UPDATE_PASSWORD_ACTION_ERROR_CODE);
+            } else {
+                throw new UserStoreException("Unknown status received from the action executor.");
+            }
```

Comment:
This state won't come. So no need to handle this. And it should of course not ideally come and a part of action execution service to handle